### ISSUE TITLE
LLaMA can't generate eos token

### ISSUE THREAD

**Issue by zetian1025** [facebookresearch/llama#321]
Hi, when I tried your models, I found that the model can't generate eos token, which means the model can't stop generation.

Do you think it's because eos token wasn't included in the pretraining stage, or simply because the generation procedure hasn't finished? (which means the eos token can be generated for some cases)

Thanks!

**Comment by yixuantt** [comment]
+1

**Comment by Nicolas99-9** [comment]
+1

**Comment by dongwang218** [comment]
This is expected for generative language models without going through instruction finetuning. There are several workaround discussed [here]( https://github.com/huggingface/transformers/issues/24994), including repetition penalty for logits sampling. Since the release of LLaMA 2, the chat models can produce eos token appropriately.
